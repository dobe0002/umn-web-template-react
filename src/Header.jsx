/* eslint-disable import/extensions */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import WordMark from './HeaderFooter/WordMark.jsx';
import MainNavLinks from './HeaderFooter/MainNavLinks.jsx';
import SearchIcon from './HeaderFooter/SearchIcon.jsx';
import SkipLinks from './HeaderFooter/SkipLinks.jsx';

/** ** Header  **** */

const HeaderWrapper = styled.header`
  position: relative;
  min-height: ${64 - 13}px;
  /*min-width: 660px;*/
  background-color: #7a0019;
  padding: 0 54px;
  padding-top: 13px;
  color: #fff;
  font-size: 14px;
  font-family: Helvetica, 'Helvetica Neue', Futura, Verdana, Geneva, sans-serif;
  @media (max-width: 609px) {
    min-height: 90px;
  }
`;
/** ** Search  **** */
const Form = styled.form`
  display: block;
  position: absolute;
  right: 54px;
  bottom: 8px;
  @media (max-width: 775px) {
    right: 17px;
  }
  @media (max-width: 609px) {
    display: block;
    right: auto;
    top: 64px;
  }
`;

const Input = styled.input`
  font-size: 14px;
  border: none;
  padding: 4px;
  width: 175px;
  @media (max-width: 674px) {
    width: 151px;
  }
`;
const Button = styled.button`
  background-color: #7a0019;
  border: none;
`;

const Header = props => {
  const { mainContentId, mainNavId } = props;
  return (
    <HeaderWrapper>
      <SkipLinks mainContentId={mainContentId} mainNavId={mainNavId} />
      <div>
        <a href='http://umn.edu/'>
          <WordMark />
        </a>
        <MainNavLinks color='#fff' />
      </div>
      <Form id='search-form' name='gs' action='https://usearch.umn.edu/'>
        <Input
          type='text'
          name='query'
          maxLength='256'
          placeholder='Search'
          aria-label='Search for'
        />

        <Button
          type='submit'
          name='submit'
          value='search'
          aria-label='Submit Search'
        >
          <SearchIcon />
        </Button>
      </Form>
    </HeaderWrapper>
  );
};

Header.propTypes = {
  mainNavId: PropTypes.string,
  mainContentId: PropTypes.string
};
Header.defaultProps = {
  mainNavId: undefined,
  mainContentId: undefined
};

export default Header;
