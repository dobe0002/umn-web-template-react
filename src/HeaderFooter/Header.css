.umnHeaderWrapperDiv {
  position: relative;
  min-height: 68px;
  background-color: #7a0019;
  padding: 0 54px;
  padding-top: 15px;
  color: #fff;
  font-size: 14px;
  font-family: Helvetica, 'Helvetica Neue', Futura, Verdana, Geneva, sans-serif;
  box-sizing: border-box;
}

@media (max-width: 609px) {
  .umnHeaderWrapperDiv {
    min-height: 100px;
  }
  .umnHeaderSearchForm {
    display: block;
    right: auto;
    top: 64px;
  }
}
@media (max-width: 775px) {
  .umnHeaderSearchForm {
    right: 17px;
  }
}

.umnHeaderSearchForm {
  display: block;
  position: absolute;
  right: 54px;
  bottom: 8px;
}

.umnHeaderSearchInput {
  font-size: 14px;
  border: none;
  padding: 4px;
  width: 175px;
}
.umnHeaderSearchButton {
  background-color: #7a0019;
  border: none;
}
