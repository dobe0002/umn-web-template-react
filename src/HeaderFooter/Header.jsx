/* eslint-disable react/no-danger */
/* eslint-disable react/prop-types */
/* eslint-disable import/extensions */
import React from 'react';
import WordMark from './icons/WordMark';
import MainNavLinks from './MainNavLinks';
import SearchIcon from './icons/SearchIcon';
import SkipLinks from './SkipLinks';
import AppBanner from './AppBanner';
// import styled from 'styled-components'; /* Not using styled-components so  styled-components isn't needed as a dependency */
// import PropTypes from 'prop-types';  /* Disabling this so prop-types isn't needed as a dependency */

const HeaderCSS = `
.umnHeaderWrapperDiv {
  position: relative;
  min-height: 68px;
  background-color: #7a0019;
  padding: 0 54px;
  padding-top: 15px;
  color: #fff;
  font-size: 14px;
  font-family: Helvetica, 'Helvetica Neue', Futura, Verdana, Geneva, sans-serif;
  box-sizing: border-box;
}

@media (max-width: 609px) {
  .umnHeaderWrapperDiv {
    min-height: 100px;
  }
  .umnHeaderSearchForm {
    display: block;
    right: auto;
    top: 64px;
  }
}
@media (max-width: 775px) {
  .umnHeaderSearchForm {
    right: 17px;
  }
}

.umnHeaderSearchForm {
  display: block;
  position: absolute;
  right: 54px;
  bottom: 8px;
}

.umnHeaderSearchInput {
  font-size: 14px;
  border: none;
  padding: 4px;
  width: 175px;
}
.umnHeaderSearchButton {
  background-color: #7a0019;
  border: none;
}
`;
const Header = (props) => {
  const { mainContentId, mainNavId, appTitle } = props;
  return (
    <header>
      <style
        dangerouslySetInnerHTML={{
          __html: HeaderCSS,
        }}
      />
      <div className='umnHeaderWrapperDiv'>
        <SkipLinks mainContentId={mainContentId} mainNavId={mainNavId} />
        <div>
          <a href='http://umn.edu/'>
            <WordMark />
          </a>
          <MainNavLinks color='#fff' />
        </div>
        <div id='umnHeaderSearchForm'>
          <form
            className='umnHeaderSearchForm'
            id='search-form'
            name='gs'
            action='https://usearch.umn.edu/'
          >
            <input
              className='umnHeaderSearchInput'
              type='text'
              name='query'
              maxLength='256'
              placeholder='Search'
              aria-label='Search for'
            />

            <button
              className='umnHeaderSearchButton'
              type='submit'
              name='submit'
              value='search'
              aria-label='Submit Search'
            >
              <SearchIcon />
            </button>
          </form>
        </div>
      </div>
      {appTitle !== undefined && appTitle !== '' && (
        <AppBanner appTitle={appTitle} />
      )}
    </header>
  );
};

// Header.propTypes = {
//   mainNavId: PropTypes.string,
//   mainContentId: PropTypes.string,
//   appTitle: PropTypes.string,
// };
// Header.defaultProps = {
//   mainNavId: undefined,
//   mainContentId: undefined,
//   appTitle: undefined,
// };

export default Header;
