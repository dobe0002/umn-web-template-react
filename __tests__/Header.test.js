import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Header from '../src/Header';
import Adapter from 'enzyme-adapter-react-16';
const { axe, toHaveNoViolations } = require('jest-axe');
expect.extend(toHaveNoViolations);

configure({ adapter: new Adapter() });

describe('UMN Header', () => {
  it('Accessibility check', async () => {
    const header = mount(<Header />);
    const results = await axe(header.getDOMNode());

    console.log('Axe violations', results.violations);

    expect(results).toHaveNoViolations();
  });
  it('Header renders correctly', () => {
    const header = shallow(<Header />);
    // console.log(header.html());
    expect(header).toMatchSnapshot();
  });
});
