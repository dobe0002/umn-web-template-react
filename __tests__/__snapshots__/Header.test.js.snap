// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UMN Header Header renders correctly 1`] = `
<styled.header>
  <SkipLinks />
  <div>
    <a
      href="http://umn.edu/"
    >
      <WordMark
        title="UofM Home page"
        width="288px"
      />
    </a>
    <MainNavLinks
      color="#fff"
      mode="header"
    />
  </div>
  <styled.form
    action="https://usearch.umn.edu/"
    id="search-form"
    name="gs"
  >
    <styled.input
      aria-label="Search for"
      maxLength="256"
      name="query"
      placeholder="Search"
      type="text"
    />
    <styled.button
      aria-label="Submit Search"
      name="submit"
      type="submit"
      value="search"
    >
      <SearchIcon />
    </styled.button>
  </styled.form>
</styled.header>
`;
